mirgecom/simutil.py

```diff
@@ -162,6 +148,40 @@ def sim_checkpoint(discr, visualizer, eos, q, vizname, exact_soln=None,
         raise ExactSolutionMismatch(step, t=t, state=q)
 
 
+def sim_healthcheck(discr, eos, q, conserved_vars, step=0, t=0):
```

Comment:
We'll have to generalize this so it's not hard-coded to fluids (or otherwise make separate versions for different physics?), but maybe we can do that in #257 along with what's already there.

Reply:
@majosm Do you want to maybe talk sometime about combining this PR with yours? I haven't actually looked at #257 in great detail, so maybe there's an opportunity to revive it?

Reply:
I suspect there will still be a fair amount of back and forth with that PR when it gets revived, so maybe it would be best to try to get a basic health check functionality into main with this PR and then we can integrate that into #257 later?

Reply:
> We'll have to generalize this so it's not hard-coded to fluids (or otherwise make separate versions for different physics?), but maybe we can do that in #257 along with what's already there.

I think health checks necessarily need to have knowledge of the operator-specific physics. Maybe not if all it does it check for NAN/INF, but otherwise... yeah.  Checkpoint is similar, as it was initially designed to be the fluid health check.  Since we have viz (awkwardly) all wrapped up in the `simutils::checkpoint`... it's awkward to generalize.  #257 is showing some promise for a path out of this :)